src/table.rs

```diff
@@ -192,6 +193,7 @@ impl Table {
     /// # Panics
     ///
     /// If `id` is out of bounds or the does not have the type `T`.
+    #[inline]
```

Comment:
This I am not sure we want to inline, all of the functions within it are already inlined so this function itself is fairly big and as such might start bloating salsa too much given its fairly frequently used

Reply:
I'll try without inlining it.

Reply:
It doesn't seem to change perf, so I removed the inline.